deisctl/cmd/cmd.go

```diff
@@ -184,13 +184,19 @@ func startDefaultServices(b backend.Backend, wg *sync.WaitGroup, outchan chan st
 	wg.Wait()
 
 	// optimization: start all remaining services in the background
+	// cache is a special case because it must start before registry
+	b.Start([]string{"cache"}, &_wg, _outchan, _errchan)
```

Comment:
The block exists as an optimization.  We start a large group of services in the background, then start services in the foreground (waiting for them to become active) so the UI makes sense.

Regarding this logic, you only need to call `wg.Wait()` the first time you start cache (on line 188).  You can leave the foreground start as it is (revert lines 197-199), since we're only waiting for services to go `active/active` at that point.